Src/Support/Google.Apis.Auth/OAuth2/ImpersonatedExternalAccountCredential.cs

```diff
@@ -0,0 +1,129 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Http;
+using System;
+using System.Collections.Generic;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    internal class ImpersonatedExternalAccountCredential : ImpersonatedCredentialBase, IExternalAccountCredential, IGoogleCredential
```

Comment:
A comment would be really welcome here, as I'm pretty confused :)

Reply:
I was wondering about `ImpersonatingExternalAccountCredential`, yes - as you say, it's not being impersonated, it's doing the impersonating.

I think it's okay given the comment though.

Reply:
This class wil also go away now, btw.